app/services/PushPullNotificationService.scala

```diff
@@ -61,4 +65,22 @@ class PushPullNotificationService @Inject()(connector: PushPullNotificationConne
           logger.error(s"Error while sending push notification", e)
       }
 
+  def sendPushNotificationIfBoxExists(
+    departure: Departure,
+    messageResponse: MessageResponse
+  )(implicit hc: HeaderCarrier, ec: ExecutionContext, request: Request[NodeSeq]): Future[Unit] =
```

Comment:
Again should the request be implicit here since we are explicitly using the `body`?

Reply:
Basically my philosophy on implicits is that normally they should either only be used by other implicits or should indicate some kind of constraint - I don't even particularly like `HeaderCarrier` being used explicitly because it breaks this rule in the other direction 😆